node/src/components/block_executor.rs

```diff
@@ -528,9 +542,23 @@ impl<REv: ReactorEventT> Component<REv> for BlockExecutor {
             Event::RunStepResult { mut state, result } => {
                 trace!(?result, "run step result");
                 match result {
-                    Ok(StepResult::Success { post_state_hash }) => {
+                    Ok(StepResult::Success {
+                        post_state_hash,
+                        next_era_validators,
+                    }) => {
                         state.state_root_hash = post_state_hash.into();
-                        self.finalize_block_execution(effect_builder, state)
+                        let next_era_validators: BTreeMap<PublicKey, U512> = next_era_validators
+                            .into_iter()
+                            .map(|(casper_types_public_key, weight)| {
+                                let public_key = PublicKey::try_from(casper_types_public_key).expect("Could not convert casper_types public key into node public key");
```

Comment:
@afck @fizyk20 @goral09 Feels bad to do an `expect` here... is there something better I can do?

Reply:
Perhaps a `filter_map` would be better here...?

Reply:
That can only fail if a validator bonds with an invalid public key, can it?
I think we have the same problem in `EraSupervisor` somewhere; we just need to make sure we do the same thing in both places.

(Also: formatting! :grin:)

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/717/commits/86eff05cf00063166109268eaaaf6e1f5436cc06